unliftio/src/UnliftIO/Exception.hs

```diff
@@ -472,6 +501,24 @@ toAsyncException e =
   where
     se = toException e
 
+-- | Convert from an asynchronous exception, the inverse of 'toAsyncException'.
+--
+-- For asynchronous exceptions, will unwrap all of the possible ways
+-- an async exception may be thrown (i.e. both unliftio's throwTo and
+-- Control.Exception's throwTo).
```

Comment:
This is slightly misleading. `Control.Exception.throwTo` doesn't perform any wrapping on exceptions. In fact, I think the body of this can possibly be simplified to:

```haskell
fromAsyncException se = do
  case fromException se of
    AsyncExceptionWrapper e -> fromException e
    Nothing -> fromException se
```

The same logic applies to `SyncExceptionWrapper`. Which makes me think it would make more sense to have a more general `fromExceptionUnwrap` that handles _both_ data types at the same time.

Reply:
Yeah, that sounds good. I was thinking of that anyway.